src/zope/component/hookable.py

```diff
@@ -11,23 +11,8 @@
 # FOR A PARTICULAR PURPOSE.
 #
 ##############################################################################
-""" This module supplies a pure-Python version of zope.hookable.hookable.
 """
-class hookable(object):
-    __slots__ = ('__original', '__implementation')
-
-    original = property(lambda self: self.__original,)
-    implementation = property(lambda self: self.__implementation,)
-
-    def __init__(self, implementation):
-        self.__original = self.__implementation = implementation
-
-    def sethook(self, newimplementation):
-        old, self.__implementation = self.__implementation, newimplementation
-        return old
-
-    def reset(self):
-        self.__implementation = self.__original
-
-    def __call__(self, *args, **kw):
-        return self.__implementation(*args, **kw)
+This module is deprecated. Prefer to use zope.hookable.hookable.
+"""
+from zope.hookable import hookable
+hookable = hookable # BBB
```

Comment:
Some sort of deprecation warning would be nice here.  Perhaps we should use zope.deprecation?

Reply:
I can if we feel like we need it. I didn't initially for two reasons:

1. I think this is internal API. At least, the module is not documented in the API docs, and the `hookable` class had no docstring.
2. I couldn't find any uses of it in a github code search (other than the copies of this project)

(Also, DeprecationWarning is silent by default, meaning that it usually gets missed. My org uses a monkey-patch to zope.deprecation to change it to use FutureWarning instead so that we don't have to mess with command line arguments or warning filters to see them. There's a *huge* discussion on python-dev about this right now.)

Reply:
I can see another reason to add zope.deprecation: There are lots of backwards compatibility exports in `zope.component.interfaces` and `zope.component.registry` that trigger linter warnings. These could all be marked with `zope.deprecation`, or, ideally, `zope.deferredimport.deprecatedFrom`. I'll attempt those things in a separate PR.